mongo/client.go

```diff
@@ -271,6 +271,9 @@ func (c *Client) Ping(ctx context.Context, rp *readpref.ReadPref) error {
 
 // StartSession starts a new session configured with the given options.
 //
+// StartSession does not actually communicate with the server and will not necessarily error if the client is
```

Comment:
If this only errors when `c.sessionPool == nil`, will that occur when a `mongo.Client` is created but not connected? If so, I think this can be more specific, noting that this will never error on a disconnected client:

```suggestion
// StartSession does not actually communicate with the server and will not error if the client is
```

Reply:
Yep! Calling `Client#Connect()` after `Client#Disconnect()` results in `ErrClientDisconnected` (phew). Yes that looks to be correct: `c.sessionPool == nil` only when a `Client` has been created but not connected. Removed "necessarily".